plenum/common/stack_manager.py

```diff
@@ -76,35 +76,45 @@ def parseLedgerForHaAndKeys(ledger, returnActive=True):
         cliNodeReg = OrderedDict()
         nodeKeys = {}
         activeValidators = set()
-        for _, txn in ledger.getAllTxn():
-            if txn[TXN_TYPE] == NODE:
-                nodeName = txn[DATA][ALIAS]
-                clientStackName = nodeName + CLIENT_STACK_SUFFIX
-                nHa = (txn[DATA][NODE_IP], txn[DATA][NODE_PORT]) \
-                    if (NODE_IP in txn[DATA] and NODE_PORT in txn[DATA]) \
-                    else None
-                cHa = (txn[DATA][CLIENT_IP], txn[DATA][CLIENT_PORT]) \
-                    if (CLIENT_IP in txn[DATA] and CLIENT_PORT in txn[DATA]) \
-                    else None
-                if nHa:
-                    nodeReg[nodeName] = HA(*nHa)
-                if cHa:
-                    cliNodeReg[clientStackName] = HA(*cHa)
-
-                try:
-                    # TODO: Need to handle abbreviated verkey
-                    verkey = cryptonymToHex(txn[TARGET_NYM])
-                except ValueError as ex:
-                    raise ValueError("Invalid verkey. Rebuild pool transactions.")
-
-                nodeKeys[nodeName] = verkey
-
-                services = txn[DATA].get(SERVICES)
-                if isinstance(services, list):
-                    if VALIDATOR in services:
-                        activeValidators.add(nodeName)
-                    else:
-                        activeValidators.discard(nodeName)
+        try:
+            for _, txn in ledger.getAllTxn():
```

Comment:
Please move the code between try-catch into a separate private method to improve readability.